UPGRADING.md

```diff
@@ -0,0 +1,37 @@
+# Upgrading Opensearch JavaScript Client
+## Upgrading from 2.x to 3.x
+Starting from 3.x, the API functions and modules of the Opensearch JavaScript are generated from the OpenSearch REST API specification. While the vast majority of the API remains the same, there are some changes that you should be aware of when upgrading from 2.x to 3.x.
+### Breaking Changes
+
+- **Removed camelCase params**: Every API function now only accepts parameters matching the OpenSearch API Spec, which are also what the OpenSearch server expects. In previous versions, some API functions would convert camelCase parameters to snake_case before sending them to the server. This is no longer the case. For example, the `client.search()` function now only accepts `filter_path`, and passing `filterPath` will result in an error from the server.
```

Comment:
```suggestion
#### Removed camelCase params

Every API function now only accepts parameters matching the OpenSearch API Spec, which are also what the OpenSearch server expects. In previous versions, some API functions would convert camelCase parameters to snake_case before sending them to the server. This is no longer the case. For example, the `client.search()` function now only accepts `filter_path`, and passing `filterPath` will result in an error from the server.
```

This is so one can link to the paragraph easily when it needs to be mentioned in, for example, bug reports.